File: src/player/providers/eleven.ts
// ElevenLabs — bring-your-own API key (stored in Windows Credential Manager,
// read only inside Rust). Synthesis is always disk-cached: re-listening never
// re-spends credits on the same sentence+voice.

import { ipc } from "../../core/ipc";
import type { SynthResult, VoiceInfo } from "../../core/types";
import type { TtsProvider } from "./provider";

let voicesCache: VoiceInfo[] | null = null;

export const elevenProvider: TtsProvider = {
  id: "eleven",
  label: "ElevenLabs",
  kind: "audio",

  async availability() {
    try {
      const has = await ipc.hasKey("eleven");
      return has ? { ok: true } : { ok: false, reason: "Add your ElevenLabs API key in Settings" };
    } catch {
      return { ok: false, reason: "Add your ElevenLabs API key in Settings" };
    }
  },

  async voices(): Promise<VoiceInfo[]> {
    if (voicesCache) return voicesCache;
    voicesCache = await ipc.elevenVoices();
    return voicesCache;
  },

  synth(voiceId: string, text: string): Promise<SynthResult> {
    return ipc.synth("eleven", voiceId, text);
  },
};

/** Settings calls this after the key changes so the list refetches. */
export function invalidateElevenVoices(): void {
  voicesCache = null;
}

File: src-tauri/src/tts/eleven.rs
// ElevenLabs synthesis + voice list. The API key is read from the Windows
// Credential Manager inside this module and NEVER leaves Rust.

use super::VoiceInfo;
use crate::error::{AppError, Result};
use std::path::Path;

/// Synthesize one sentence to MP3 at `out`. Model comes from the settings
/// field `elevenModel` (settings::read_field_str) with a sensible default.
pub fn synth(voice_id: &str, text: &str, out: &Path) -> Result<()> {
    let _ = (voice_id, text, out);
    Err(AppError::msg("not implemented yet"))
}

#[tauri::command]
pub async fn eleven_voices() -> Result<Vec<VoiceInfo>> {
    Err(AppError::msg("not implemented yet"))
}
